src/H5Tconv.c

```diff
@@ -3801,6 +3818,57 @@ H5T__conv_vlen(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata, const H5T
     } /* end switch */
 
 done:
+    /* Release converted elements on error */
+    if (ret_value < 0 && conversions_made) {
+        size_t dest_count;
+
+        /* Set up for first pass to destroy references */
+        if (nelmts < orig_nelmts || (convert_forward && elmtno < safe)) {
+            dest_count = orig_nelmts - nelmts;
+
+            /* Set pointer to correct location, based on direction chosen */
+            if (convert_forward) {
+                d = (uint8_t *)buf;
+                dest_count += elmtno; /* Include partial iteration in first pass, for forward conversions */
+            }
+            else
+                d = (uint8_t *)buf + (nelmts * orig_d_stride);
+
+            /* Destroy references that have already been converted */
```

Comment:
Nitpick: this is for freeing variable-length elements rather than destroying references. Same for below

Reply:
Fixed